### DISCUSSION TITLE
Generate typescript types for your frontend

### DISCUSSION THREAD

**Discussion by patmood** [pocketbase/pocketbase#773] [Show and tell]
I created a small package to generate types from the pocketbase db:

https://github.com/patmood/pocketbase-typegen

I plan on using this in my front end projects so that records can be accurately typed and up to date after changes to the database schema.

Hope this is useful to someone. I'll keep developing it as I find ways to improve.

**Comment by benallfree** [comment]
Hi @patmood, could you please update the readme with sample output? I think that would really help people understand.

**Reply by patmood** [reply]
Done, good idea!

**Comment by patmood** [comment]
@ganigeorgiev any interest in allowing users to pass in a record type to the SDK? For example: `client.records.getList<ProfileRecord>()` 

Where `ProfileRecord` comes from this package (or the user's own manual definition). This would allow the responses to be properly typed, rather than casting it.

**Reply by ganigeorgiev** [reply]
@patmood This is already implemented in the new v0.8.x version of the SDK in relation to issue https://github.com/pocketbase/js-sdk/issues/38.

_It will be available soon. Expect a pre-release within the next 2-3 weeks._

**Reply by patmood** [reply]
That's great to hear! Keep up the great work 🙌

**Comment by benallfree** [comment]
@patmood That sample output is great! Thanks so much.

Now the only other thing I think would be SUPER helpful is if you could use the export JSON rather than having to directly reference the SQLite file. Some developers may not have easy access to that file. For example, pockethost.io users have admin access to their instance but not file access.

In any case, this is great and I'll add it to [awesome-pocketbase](https://github.com/benallfree/awesome-pocketbase).

**Reply by patmood** [reply]
Sounds good! I'll add that and also a way to call a prod api directly to save exporting

**Reply by patmood** [reply]
Just updated it to handle JSON and URL options and verified with pockethost. Nice little service!

**Comment by mapsgeek** [comment]
Thank you <3

**Comment by babakfp** [comment]
Gen:

```ts
export enum TiersVisibilityOptions {
	"public" = "public",
	"private" = "private",
}
export type TiersRecord = {
	invites?: number
	name: string
	price?: number
	videos?: RecordIdString[]
	visibility: TiersVisibilityOptions
}
```

My code:

```ts
const tiers = await locals.pb.collection("tiers").getFullList<
    TiersResponse & {
        visibility: Exclude<
            TiersVisibilityOptions,
            TiersVisibilityOptions.private
        >
    }
>({ filter: 'visibility = "public"' })
```

It would be nice to not need to do this part:

```ts
<
    TiersResponse & {
        visibility: Exclude<
            TiersVisibilityOptions,
            TiersVisibilityOptions.private
        >
    }
>
```

**Comment by babakfp** [comment]
I remember there was a conversation about improving PocketBase type-safety, but I couldn't find it. Can someone send it to me, please?
